include/mysqlpp/udf_wrappers.hpp

```diff
@@ -124,6 +125,14 @@ class udf_base {
       message[message_size - 1] = '\0';
     }
   }
+  static void validate_argument_nullness(const udf_context &udf_ctx) {
+    for (std::size_t index = 0; index < udf_ctx.get_number_of_args(); ++index) {
+      if (!udf_ctx.is_arg_nullable(index) && udf_ctx.is_arg_null(index)) {
+        throw std::invalid_argument("argument " + std::to_string(index) +
```

Comment:
In other error messages which mention argument number (ER_SP_NOT_VAR_ARG, ER_INVALID_JSON_TEXT_IN_PARAM, ER_INVALID_TYPE_FOR_JSON) we start counting arguments from 1 and not 0 like you do here. I think it is better to be consistent with server and do the same thing for UDFs as well.

Reply:
You are 100% right. This is a good point. Users will expect error messages to have argument numbers starting with `1`.

Reply:
Code fixed,  MTR test cases re-recorded.

Reply:
Great!